### PR TITLE
feat(new-tasks): Fetch task collaborators

### PR DESCRIPTION
**Pull request by shahzadaziz**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @shahzadaziz has signed the CLA. Thanks for the pull request!

File: src/elements/content-sidebar/activity-feed/task-new/Task.js [hunk 1]
```diff
@@ -105,9 +123,43 @@ class Task extends React.Component<Props, State> {
...
+        const { id, api, task_links } = this.props;
+        const { entries } = task_links;
+        let fileId;
+
+        if (entries[0].target) {
+            fileId = entries[0].target.id;
+        }
+        this.setState({ isLoading: true });
+
+        return new Promise((resolve, reject) => {
```
Comment (priyajeet) [inline src/elements/content-sidebar/activity-feed/task-new/Task.js]: You have async above, but no await usage, should convert to use await

File: src/elements/content-sidebar/activity-feed/task-new/Task.js [hunk 2]
```diff
@@ -105,9 +123,43 @@ class Task extends React.Component<Props, State> {
         );
     };
 
+    fetchTaskCollaborators = async () => {
+        const { id, api, task_links } = this.props;
+        const { entries } = task_links;
+        let fileId;
+
+        if (entries[0].target) {
+            fileId = entries[0].target.id;
```
Comment (alexkrolick) [inline src/elements/content-sidebar/activity-feed/task-new/Task.js]: Do we need to return early if there is no fileId?

File: src/elements/content-sidebar/activity-feed/task-new/Task.js (hunk 1 above)
Comment (alexkrolick) [reply src/elements/content-sidebar/activity-feed/task-new/Task.js]: API methods like `getTaskCollaborators` do not return a Promise, this wraps the callbacks to enable Promise chaining or async/await elsewhere in the component

File: src/elements/content-sidebar/activity-feed/task-new/Task.js (hunk 2 above)
Comment (shahzadaziz) [reply src/elements/content-sidebar/activity-feed/task-new/Task.js]: Fixed

File: src/elements/content-sidebar/activity-feed/task-new/Task.js (hunk 1 above)
Comment (shahzadaziz) [reply src/elements/content-sidebar/activity-feed/task-new/Task.js]: Removed async from signature.

File: src/elements/content-sidebar/activity-feed/task-new/__tests__/Task-test.js [hunk 3]
```diff
@@ -251,4 +256,33 @@ describe('elements/content-sidebar/ActivityFeed/task-new/Task', () => {
...
+                currentUser={currentUser}
+                error={{ title: 'blah', message: 'blah' }}
+                onEdit={jest.fn()}
+                onDelete={jest.fn()}
+            />,
+        );
+        const instance = wrapper.instance();
+        instance.fetchTaskCollaborators = jest
+            .fn()
+            .mockRejectedValueOnce()
```
Comment (alexkrolick) [inline src/elements/content-sidebar/activity-feed/task-new/__tests__/Task-test.js]: This looks like it is triggering the error state

File: src/elements/content-sidebar/activity-feed/task-new/__tests__/Task-test.js [hunk 4]
```diff
@@ -251,4 +256,33 @@ describe('elements/content-sidebar/ActivityFeed/task-new/Task', () => {
...
+            .mockResolvedValueOnce({});
+
+        await instance.handleEditClick();
+
+        expect(instance.fetchTaskCollaborators).toBeCalled();
```
Comment (alexkrolick) [inline src/elements/content-sidebar/activity-feed/task-new/__tests__/Task-test.js]: Can we go one step further in the tests and check that the fetched assignees should up where they should?

Comment (shahzadaziz) [reply src/elements/content-sidebar/activity-feed/task-new/__tests__/Task-test.js]: Working on it. I can add some tests as a follow up PR

**Pull request merged**